python/cuml/test/test_random_forest.py

```diff
@@ -550,3 +552,47 @@ def test_rf_regression_sparse(datatype, split_algo, mode, column_info,
         sk_r2 = r2_score(y_test, sk_predict, convert_dtype=datatype)
         assert fil_r2 >= (sk_r2 - 0.07)
     assert fil_r2 >= (cu_r2 - 0.02)
+
+
+@pytest.mark.parametrize('datatype', [np.float32])
+@pytest.mark.parametrize('column_info', [unit_param([20, 10]),
+                         quality_param([200, 100]),
+                         stress_param([500, 350])])
+@pytest.mark.parametrize('nrows', [unit_param(500), quality_param(5000),
+                         stress_param(500000)])
+def test_rf_memory_leakage(datatype, column_info, nrows):
+    n_iter = 100
+    use_handle = True
+    ncols, n_info = column_info
+    X, y = make_classification(n_samples=nrows, n_features=ncols,
+                               n_clusters_per_class=1, n_informative=n_info,
+                               random_state=0, n_classes=2)
+    X = X.astype(datatype)
+    y = y.astype(np.int32)
+    X_train, X_test, y_train, y_test = train_test_split(X, y, train_size=0.8,
+                                                        random_state=0)
+    # Create a handle for the cuml model
+    handle, stream = get_handle(use_handle, n_streams=1)
+
+    # Warmup. Some modules that are used in RF allocate space on the device
+    # and consume memory. This is to make sure that the allocation is done
+    # before the first call to get_memory_info.
+    base_model = curfc(handle=handle)
+    base_model.fit(X_train, y_train)
+
+    free_mem = cuda.current_context().get_memory_info()[0]
+
+    rfc_model = curfc(handle=handle)
+    rfc_model.fit(X_train, y_train)
+
+    # Calculate the memory free after fitting the cuML RF model
+    delta_mem = free_mem - cuda.current_context().get_memory_info()[0]
+
+    for i in range(n_iter):
+        cuml_mods = curfc(handle=handle)
+        cuml_mods.fit(X_train, y_train)
```

Comment:
Do you need to refit each time? seems like we could make the test run faster by just fitting once and calling predict in a loop

Reply:
yes, we need to since rf does not let us fit over a model that has already been trained. We cannot run fit over an exsisting forest.

Reply:
I was suggesting not running fit again but just running predict over and over

Reply:
ohh sorry, i misunderstood, i can just run predict and check the memory leak